### PR TITLE
[WIP] :seedling: Drop MachinePools from topology self-hosted tests

### PR DESCRIPTION
**Pull request by killianmuldoon**
Creating this WIP PR as a template of what should be merged into the `release-1.6` branch - due to be created on November 14th - in order to complete https://github.com/kubernetes-sigs/cluster-api/issues/9656.

Fixes https://github.com/kubernetes-sigs/cluster-api/issues/9656

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Note: This change should only be made on `release-1.6`. Someone will have to open a new PR with the same contents against that branch.


/hold 

/cc @kubernetes-sigs/cluster-api-release-team @nawazkh

**Comment by killianmuldoon** [comment]
@adilGhaffarDev

File: test/framework/cluster_topology_helpers.go [hunk 1]
```diff
@@ -84,8 +84,6 @@ func UpgradeClusterTopologyAndWaitForUpgrade(ctx context.Context, input UpgradeC
 	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling UpgradeClusterTopologyAndWaitForUpgrade")
 	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling UpgradeClusterTopologyAndWaitForUpgrade")
 	Expect(input.ControlPlane).ToNot(BeNil(), "Invalid argument. input.ControlPlane can't be nil when calling UpgradeClusterTopologyAndWaitForUpgrade")
-	Expect(input.MachineDeployments).ToNot(BeEmpty(), "Invalid argument. input.MachineDeployments can't be empty when calling UpgradeClusterTopologyAndWaitForUpgrade")
```
Comment (killianmuldoon) [inline test/framework/cluster_topology_helpers.go:87]: This is a significant change in this PR, but I think we should allow upgrades of Clusters that don't have these workers defined in the yamls.

The alternative is to fill these lists in as empty at some point if they are nil, but it seems like there's little difference between the two at this point.

**Comment by CecileRobertMichon** [comment]
We shouldn't need to do this now that #8842 has merged. Let's keep monitoring the signal in the next few days and close this PR if we don't see the flake anymore.

**Comment by CecileRobertMichon** [comment]
/close

flakes are resolved

**Pull request closed without merging**